File: benchmarks/competitors/bench_loop.py
#!/usr/bin/env python3
# ==============================================================================
# 🐍 PYTHON BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.py)
# ==============================================================================

import time

start = time.perf_counter()
total = 0
i = 1
while i <= 10000000:
    total += i % 7
    i += 1

elapsed_ms = (time.perf_counter() - start) * 1000.0
print("Python Loop 10,000,000 Iterations Complete!")
print(f"Sum: {total}")
print(f"Elapsed Time: {elapsed_ms:.2f} ms")

File: benchmarks/competitors/bench_loop.cs
// ==============================================================================
// 🔷 C# BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.cs)
// ==============================================================================

using System;
using System.Diagnostics;

class Program
{
    static void Main()
    {
        var sw = Stopwatch.StartNew();

        ulong total = 0;
        for (ulong i = 1; i <= 10000000; i++)
        {
            total += (i % 7);
        }

        sw.Stop();
        double elapsedMs = sw.Elapsed.TotalMilliseconds;

        Console.WriteLine("C# Loop 10,000,000 Iterations Complete!");
        Console.WriteLine($"Sum: {total}");
        Console.WriteLine($"Elapsed Time: {elapsedMs:F2} ms");
    }
}
